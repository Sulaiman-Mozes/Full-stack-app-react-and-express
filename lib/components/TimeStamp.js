import React, { Component } from 'react';
import PropTypes from 'prop-types';

class TimeStamp extends Component {
  shouldComponentUpdate(nextProps) {
    const { timestamp } = this.props;
    return (
      this.timestampDisplay(timestamp)
      !== this.timestampDisplay(nextProps.timestamp));
  }

  timestampDisplay = timestamp => timestamp.toLocaleTimeString('en-US',
    { hour: '2-digit', minute: '2-digit', hour12: true });

  render() {
    const { timestamp } = this.props;
    return (
      <div>
        Current Time
        {' '}
        {this.timestampDisplay(timestamp)}
      </div>
    );
  }
}

TimeStamp.propTypes = {
  timestamp: PropTypes.instanceOf(Date).isRequired,
};

export default TimeStamp;
