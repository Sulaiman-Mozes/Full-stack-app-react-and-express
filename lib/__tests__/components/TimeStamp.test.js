import React from 'react';
import { shallow } from 'enzyme';
import TimeStamp from '../../components/TimeStamp';


describe('Render Component', () => {
  it('It should render TimeStamp Component', () => {
    const timestamp = new Date();
    const wrapper = shallow(<TimeStamp timestamp={timestamp} />);
    expect(wrapper).toMatchSnapshot();
  });
});
