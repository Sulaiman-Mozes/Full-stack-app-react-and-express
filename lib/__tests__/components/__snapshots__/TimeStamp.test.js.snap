// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Component It should render TimeStamp Component 1`] = `
<div>
  5:39 PM
</div>
`;
